### DISCUSSION TITLE
How do I modify a text box with 5.x.x?

### DISCUSSION THREAD

**Discussion by ironiris** [Palakis/obs-websocket#1119] [Q&A]
in protocol version 4.9.1(windows), I was modifying it like this,
`            obs.send('SetTextGDIPlusProperties', {
              'source': 'text box name',
              'text': '0'
            })
`
How should I modify it after version 5.0.0?

**Comment by CloudMax94** [comment]
I came across the same issue with `SetTextFreetype2Properties` today and couldn't find any helpful documentation but eventually managed to guess my way to success. 

Seems they now have a generic approach to assigning any setting value to any source. Try the snippet below, it worked with `SetTextFreetype2Properties` at least.
```js
obs.call("SetInputSettings", {
  inputName: "text box name",
  inputSettings: {
    text: "0",
  },
});
```

**Reply by PKevin123** [reply]
I spent the whole afternoon on this problem, almost gave up, your method worked, thanks man!

**Reply by Atomme1** [reply]
Litteraly saved my project XD
Thank you so much <3 !

**Comment by jeanbatfr** [comment]
Hi, I'm trying to do the same thing but by sending a json with node-red, i tried to understand the documentation, tried several things like : 
{
    "op": 6,
    "d": {
        "requestType": 'SetInputSettings',
        "requestData": {
            "inputName": 'PV1_textbox',
            "overlay ": true
        },
        "inputSettings": {
            "text": "test"
        }
    }
}

or :  
{
    op: 6,
    d: {
        requestType: "SetInputSettings",
        inputName: "PV1_textbox", 
        inputSettings: {
            text: "test"  // Le texte que tu veux afficher
        }
    }
}

But nothing works.
Please help, I don't understand the doc, what is the requestID?? 

Thank you

**Reply by Atomme1** [reply]
I don’t know the language you want to use for your project but when it comes to communicate with the obs WS, you have to give it a certain amount of data.

I can’t give you precise indications because i did my project a while ago, but I you can still go and check it on my profile.
It’s called O-WA (short for obs web app)
It’s a restful api in python using flask as backend exposing a webpage that you can interact with.

The functions used to make calls to the websocket are stored in a specific python file.
